src/omv/py/py_sensor.c

```diff
@@ -101,6 +101,45 @@ static mp_obj_t py_sensor_snapshot(uint n_args, const mp_obj_t *args, mp_map_t *
     return image;
 }
 
+static mp_obj_t py_sensor_snapshot_start()
+{
+#if MICROPY_PY_IMU
+    // +-10 degree dead-zone around pitch 90/270.
+    // +-35 degree active-zone around roll 0/90/180/270/360.
+    do_auto_rotation(10, 35);
+    // We're setting the dead-zone on pitch because roll readings are invalid there.
+    // We're not setting the full range on roll to prevent oscillation.
+#endif // MICROPY_PY_IMU
+
+    int ret = sensor.snapshot_start(&sensor);
+
+    if (ret < 0) {
+        nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_RuntimeError, "Capture Start Failed: %d", ret));
+    }
+
+    return mp_const_none;
+}
+
+static mp_obj_t py_sensor_snapshot_check()
+{
+    bool status;
+    sensor.snapshot_check(&sensor, &status);
```

Comment:
Make this just return the bool. Then it's one line.

Reply:
I made commit 3f1d58d that makes these changes for you but I can't really test the stream callback